test/src/main/java/org/apache/accumulo/test/functional/AmpleConditionalWriterIT.java

```diff
@@ -382,6 +397,125 @@ public void testSelectedFiles() throws Exception {
+      assertEquals(selectedFiles, context.getAmple().readTablet(e1).getSelectedFiles(),
+          "Selected files should match those that were written");
+
+      final Text row = e1.toMetaRow();
+      final Text selectedColumnFamily = SELECTED_COLUMN.getColumnFamily();
+      final Text selectedColumnQualifier = SELECTED_COLUMN.getColumnQualifier();
+
+      Supplier<String> selectedMetadataValue = () -> {
+        try (Scanner scanner = client.createScanner(MetadataTable.NAME)) {
+          scanner.fetchColumn(selectedColumnFamily, selectedColumnQualifier);
+          scanner.setRange(new Range(row));
+
+          return scanner.stream().map(Map.Entry::getValue).map(Value::get)
+              .map(entry -> new String(entry, UTF_8)).collect(onlyElement());
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      };
+
+      String actualMetadataValue = selectedMetadataValue.get();
+      final String expectedMetadata = selectedFiles.getMetadataValue();
+      assertEquals(expectedMetadata, actualMetadataValue,
+          "Value should be equal to metadata of SelectedFiles object that was written");
+
+      // get the string value of the paths
+      List<String> filesPathList = storedTabletFiles.stream().map(StoredTabletFile::toString)
+          .sorted().collect(Collectors.toList());
+
+      // verify we have the format of the json correct
+      String newJson = createSelectedFilesJson(fateTxId, initiallySelectedAll, filesPathList);
+      assertEquals(actualMetadataValue, newJson,
+          "Test json should be identical to actual metadata at this point");
+
+      // reverse the order of the files and create a new json
+      Collections.reverse(filesPathList);
+      newJson = createSelectedFilesJson(fateTxId, initiallySelectedAll, filesPathList);
+      assertNotEquals(actualMetadataValue, newJson,
+          "Test json should have reverse file order of actual metadata");
+
+      // write the json with reverse file order
+      try (BatchWriter bw = client.createBatchWriter(MetadataTable.NAME)) {
+        Mutation mutation = new Mutation(row);
+        mutation.put(selectedColumnFamily, selectedColumnQualifier,
+            new Value(newJson.getBytes(UTF_8)));
+        bw.addMutation(mutation);
+      }
+
+      // verify the metadata has been changed
+      actualMetadataValue = selectedMetadataValue.get();
+      assertEquals(newJson, actualMetadataValue,
+          "Value should be equal to the new json we manually wrote above");
+
+      // submit a mutation with the condition that the selected files match what was originally
+      // written
+      TabletMetadata tm1 =
+          TabletMetadata.builder(e1).putSelectedFiles(selectedFiles).build(SELECTED);
+      ctmi = new ConditionalTabletsMutatorImpl(context);
+      ctmi.mutateTablet(e1).requireAbsentOperation().requireSame(tm1, SELECTED).deleteFile(stf1)
```

Comment:
Instead of doing `.deleteFile(stf1)` something could be added or set on the tablet metadata.  Then later after verifying the mutation was accepted that could be checked to ensure the change is there.

Reply:
Changed this to `putFile()` in c21b7c4